Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs

```diff
@@ -538,24 +538,26 @@ private bool IsWarning(string line)
         {
             internal readonly Predicate<string> LinePredicate;
             internal readonly Regex Regex;
+            internal readonly string GroupName;

```

Comment:
>internal readonly string GroupName; [](start = 12, length = 35)

Add comment that this `GroupName` is only associated with the `Regex`, and not `LinePredicate`. This association needs to be reflected in the contract as well.

Reply:
GroupName can currently only be added via a constructor that sets a regex, which is checked by the contract.  If a value is not passed, it already defaults to the previous hardcoded value that's already assumed by the existing callers.  Comment added about GroupName.